homeassistant/components/actronair/coordinator.py

```diff
@@ -0,0 +1,103 @@
+"""Coordinator for Actron Air Neo integration."""
+
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Any
+
+from actron_neo_api import (
+    ActronAirNeoACSystem,
+    ActronAirNeoStatus,
+    ActronNeoAPI,
+    ActronNeoAPIError,
+    ActronNeoAuthError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import dt as dt_util
+
+from .const import _LOGGER
+
+STALE_DEVICE_TIMEOUT = timedelta(hours=24)
+ERROR_NO_SYSTEMS_FOUND = "no_systems_found"
+ERROR_UNKNOWN = "unknown_error"
+
+
+@dataclass
+class ActronNeoRuntimeData:
+    """Runtime data for the Actron Air Neo integration."""
+
+    api_client: "ActronNeoApiClient"
+    system_coordinators: dict[str, "ActronNeoSystemCoordinator"]
+
+
+type ActronNeoConfigEntry = ConfigEntry[ActronNeoRuntimeData]
+
+AUTH_ERROR_THRESHOLD = 3
+SCAN_INTERVAL = timedelta(seconds=30)
+
+
+class ActronNeoApiClient:
+    """Client for Actron Neo API."""
+
+    def __init__(self, hass: HomeAssistant, entry: ActronNeoConfigEntry) -> None:
+        """Initialize the client."""
+        self.hass = hass
+        self.entry = entry
+        self.api = ActronNeoAPI(refresh_token=entry.data[CONF_API_TOKEN])
+        self.systems: list[ActronAirNeoACSystem] = []
+
+    async def async_setup(self) -> bool:
+        """Perform initial setup, including refreshing the token."""
+        try:
+            self.systems = await self.api.get_ac_systems()
+            await self.api.update_status()
+        except ActronNeoAuthError:
+            _LOGGER.error(
+                "Authentication error while setting up Actron Neo integration"
+            )
+            raise
+        except ActronNeoAPIError as err:
+            _LOGGER.error("API error while setting up Actron Neo integration: %s", err)
+            raise
+        return True
```

Comment:
Okay so this is just an object that fetches info and has no other tasks. In the future, this might become its own coordinator. I think we can just move this code to async setup entry in `__init__.py`, as there is no real need for this to be an object